src/Build/Construction/Solution/ProjectInSolution.cs

```diff
@@ -110,6 +110,12 @@ public sealed class ProjectInSolution
         private readonly Dictionary<string, ProjectConfigurationInSolution> _projectConfigurations;
         private IReadOnlyDictionary<string, ProjectConfigurationInSolution> _projectConfigurationsReadOnly;
 
+        /// <summary>
+        /// A list of strings representing relative paths from the solution file to loose items.
+        /// </summary>
+        private readonly List<string> _solutionItems;
+        private IReadOnlyList<string> _solutionItemsAsReadOnly;
```

Comment:
Why do we need an AsReadOnly version? Seems like _solutionItems is the source of truth, and I don't think this is expected to need to be thread-safe or anything.

Reply:
I was mostly copying the existing Dependency list's implementation to prevent stylistic differences, since every codebase has different rules, etc.